src/journal/journalctl.c

```diff
@@ -2280,20 +2280,20 @@ static int run(int argc, char *argv[]) {
         }
 
         if (arg_directory)
-                r = sd_journal_open_directory(&j, arg_directory, arg_journal_type);
+                r = sd_journal_open_directory(&j, arg_directory, arg_journal_type | SD_JOURNAL_FAST_QUERY);
```

Comment:
I think here we should not set the flag unconditionally, as the specified directory may be /var/build/journal/remote.

Reply:
What is `/var/build/journal/remote` and why it should be different during querying by `journalctl`?
I don't see any references in the code to understand the issue.
Can you help me?

Reply:
sorry, I have no idea why I include 'build'...
The directory I wanted to say is `/var/log/journal/remote`.

Reply:
Hmm. I completely forget why I thought so. Feel free to ignore.
I will re-review later.